### DISCUSSION TITLE
Livewire taking too much time in browser server sends response in ms but browser take more 20s to 60s

### DISCUSSION THREAD

**Discussion by anasmansoormalik** [livewire/livewire#8270] [1. Help]
I have a data table with dynamic filters. The issue is that the server sends a response back to the browser in milliseconds, but the loading element with spinners keeps loading and takes too much time. By checking performance in Chrome DevTools, the following results are given below. Please note that the situation is better in Firefox, but worse in Chrome. 
The recording below is from Chrome:

https://github.com/livewire/livewire/assets/86790713/8d7c8cfd-a8fb-4291-89a0-717663276ede

Livewire version v3
Laravel 10

**Comment by Orteko** [comment]
I have an extremely similar issue - a datatable with a set of filters, which gets steadily worse on performance depending on the size of the result set.

The resulting data is returned to the browser in under a second, however it can then spend anywhere from 7 seconds to a minute in 'Run Microtasks' and then a series of nested (anonymous) -> diff tasks which i assume is the DOM diff/morph.

Refreshing the page so that the selected option is populated from the url instead of via a livewire update and the response is essentially instant.

The actual output is just a table which in the worst case has about two thousand rows.

Each row has a unique `wire:key` so that's not the issue.

It looks like `wire:replace` was added in https://github.com/livewire/livewire/pull/8403 as of v3.5.0 which i attempted to use figuring that might solve my issue however the DOM diffing still seemed to occur anyway and take the same amount of time which seemed odd.

@samlev unless i'm missing something, shouldn't adding a `wire:replace` on the root div of a livewire component essentially mean that no morphing should take place at all and it would always just be entirely replaced?

https://github.com/filamentphp/filament/issues/9304 is tangentially related in so much that they have a similar problem and use livewire, but i'm not using filament in my system.

**Reply by samlev** [reply]
No, I don't think that it will work on the _root_ element - there's a bunch of state attached to the root element, and you don't really want to replace it.

`wire:replace` still inspects the DOM so that the any attribute changes to the _current element_ are included, but `wire:replace.self` just replaces the entire old element with the new one, and skips any other crawling of the tree for that element/it's children.

The way I use it is on the element that wraps the `@foreach`, and `wire:replace.self` is "faster" for this:

```html
<table>
  <thead>
    <tr>
      <th>Name</th>
      <th>Email</th>
      <th>
        <x-form.checkbox wire:model.live="trashed">
          Include Disabled
        </x-form.checkbox>
      </th>
    </tr>
  </thead>
  <tbody wire:replace.self>
    @foreach($users as $user)
      <tr>
        <td>
          {{ $user->name }}
        </td>
        <td>
          {{ $user->email }}
        </td>
        <td>
          <x-users.actions :user="$user" />
        </td>
      </tr>
    @endforeach
  </tbody>
  <tfoot>
    <tr>
      <td colspan="3">
          {{ $users->links }}
      </td>
    </tr>
  </tfoot>
</table>
```

**Reply by Orteko** [reply]
Hmm yeah ok, makes sense not to use it on the root element - didn't seem to have much effect when i did that.

I did first try it on the tbody with wire:replace.self like your example but still seemed to end up having issues.

I'll see if i can replicate it in a bit more of a standalone fashion by removing all the filters etc to see if this is actually an issue with `wire:replace` or if i've got something else going on.

**Comment by silvershades** [comment]
are you using docker?

**Reply by Orteko** [reply]
No docker involved in my scenario - although i'm not sure why it would be relevant given that both @anasmansoormalik and my issue appears to be at the JS clientside of things.
